src/URLTranslator.php

```diff
@@ -24,12 +25,12 @@ public static function getUserEditURL(Member $member): string
         return self::getAdminURL() . "/security/users/EditForm/field/users/item/{$member->ID}/edit";
     }
 
-    public static function getPageEditURL(SiteTree|int $siteTreeOrID): string
+    public static function getPageEditURL(SiteTree|int $siteTree): string
     {
-        if (!is_int($siteTreeOrID)) {
-            $siteTreeOrID = $siteTreeOrID->ID;
+        if (is_int($siteTree)) {
+            $siteTree = Versioned::get_by_stage(SiteTree::class, 'Stage')->byID($pageId);
```

Comment:
At this point you do not validate whether `$siteTree` is on object, potentially breaking the followup `siteTree->CMSEditLink()` call.